src/movies.js

```diff
@@ -1,23 +1,65 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const directorList = movies.map(element => {return element.director});
+  return directorList
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(movies) {
+  if ( movies.length === 0) {
+    return 0;
+  } 
+  const spielbergDrama = movies.filter(element => { 
+    return element.director === 'Steven Spielberg' && element.genre.includes('Drama');
+  });
+return spielbergDrama.length
+} 
+    
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if ( movies.length === 0) {
+    return 0; }
+  const average = movies.reduce((acc, value) => {
+    return acc + value.score;
+    },0); 
+
+let scoreResult = average / movies.length
+return scoreResult
+}
```

Comment:
to solve this "should be rounded to 2 decimals places"
you can do this: `let scoreResult = Number((average / movies.length).toFixed(2))`

Reply:
to solve this "should return average even if one of the movies does not have score"
you can do this: 
```
const average = movies.reduce((acc, value) => {
    if (value.score){
      return acc + value.score;
    }
    else {
     return acc
    }
```